onmt/modules/multi_headed_attn.py

```diff
@@ -491,41 +491,42 @@ def forward(
             self.flash2
             and l > 256  # https://github.com/Dao-AILab/flash-attention/issues/591
         )
+        if False:
 
-        if (
-            self.max_relative_positions in [-1, 0]
-            and not return_attn
-            and query.device != torch.device("cpu")
-        ):
-            causal = self.is_decoder and self.attn_type == "self" and mask is not None
-            if self.is_decoder and self.attn_type == "self" and flash2:
-                if causal:
-                    window_size = (
-                        (-1, -1) if sliding_window == 0 else (sliding_window, 0)
-                    )
+            if (
+                self.max_relative_positions in [-1, 0]
+                and not return_attn
+                and query.device != torch.device("cpu")
+            ):
+                causal = self.is_decoder and self.attn_type == "self" and mask is not None
```

Comment:
add "and step is None" then causal will be true only for training
that will force at inference to use only the pad mask
maybe need to double check there is no side effect in NMT